### DISCUSSION TITLE
Help setting up systemd in crouton.

### DISCUSSION THREAD

**Discussion by Realmbird** [dnschneid/crouton#4998] [Q&A]
So many different things like snapd, PostgreSQL, systemctl don't seem to work since it can't access the systemd from the Chroot environment giving errors such as chroot blocked it how do I access systemd. I want to be able to use PostgreSQL and snapd. I am not that experienced using linux so maybe I might not be able to correctly understand what I am asking. I have looked at solutions all I see is "Tom Travis" github repo, do I have to uninstall crouton and reinstall his crouton not sure what I need to do to access systemd in the command line. The problem is I can't use systemctl which is dependent on systemd, and then I noticed other applications that require systemd all don't work. Is there a way to change the systemd into the default location so that I won't experience weird systemd related errors when installing things.

I have tried other things to solve them myself but they didn't fix anything. When I tried https://superuser.com/questions/688733/start-a-systemd-service-inside-chroot-from-a-non-systemd-based-rootfs service was not found.
I tried this https://github.com/smaknsk/servicectl
I tried installing servicectl then discovered that systemd was not installed 
I don't think I have enough knowledge to fix this problem browsing the web like I normally do, just don't have a better computer than chromebook.

**Comment by Qwer-TeX** [comment]
@Realmbird

I'm glad you asked. This has happened to me many times before. Basically long story short, I did some research and found out that systemd cannot run inside a chroot in this case the crouton chroot. I hope you've noticed that when installing snapd, and PostgreSQL you see something like this `Running in chroot, ignoring request`. Yes It is ignoring the request to configure snapd or PostgreSQL through systemctl. Crouton runs through chroot. And also, crouton doesn't install systemd in the installation process, instead it installs something else probably initrd, systemVinit, or upstart( used on old versions of ubuntu uptil 15.04). 

It is a pitty that we both don't have a computer better than our chromebook. 
Theres only one solution: Make a live bootable ubuntu usb stick, install custom legacy bios on your chromebook (Tiancore), and move on from there. I use a micro SD card to boot. Or you can erase the chromebook and install ubuntu on the chromebook but please note that it is highly recommended to make a backup of chromeOs through the [Chromebook Recovery Utility ](https://chrome.google.com/webstore/detail/chromebook-recovery-utili/pocpnlppkickgojjlmhdmidojbmbodfm?utm_source=chrome-app-launcher) before erasing your chromebook hard drive and install ubuntu on it.

**Also repost this in the issues tab so that the crouton developers can grab attention and give a better answer.**

**Reply by Realmbird** [reply]
So, I literally have to install Linux on my computer? no convent method of using chroot. Going to post this on issues tab since making a live bootable ubuntu stick doesn't seem like my first option. I need to first lookup how to make a live bootable stick, online.

**Comment by DennisLfromGA** [comment]
@Qwer-TeX 

All good advice except for the following, maybe clarify it a tad:

> Or you can erase the chromebook and install ubuntu on the chromebook but please note that **it is highly recommended to make a backup of chromeOs through the [Chromebook Recovery Utility ](https://chrome.google.com/webstore/detail/chromebook-recovery-utili/pocpnlppkickgojjlmhdmidojbmbodfm?utm_source=chrome-app-launcher)** before erasing your chromebook hard drive and install ubuntu on it.

The Chromebook Recovery Utility extension is needed to install a recovery image for your device but it does not perform a backup.

You make a backup of the OEM firmware using mr.chromebox's '[firmware utility script](https://github.com/MrChromebox/scripts/blob/master/firmware-util.sh)' BEFORE you erase the Chromebook.
Then, if you decide to go back to chromeOS you use the firmware utility script to restore the OEM firmware, then use the Chromebook Recovery Utility extension to reinstall chromeOS for that device.

*Hope this helps,*
-**DennisLfromGA**

**Reply by Qwer-TeX** [reply]
@DennisLfromGA 

I see. Thanks for clarifying it up :) Helps me and other people.

**Reply by Realmbird** [reply]
I looked more into its firmware utility I am thinking I might have to install Linux. Is there a crouton version that uses systemd-nspawn since it makes a container that can use systemd. https://wiki.archlinux.org/title/Systemd-nspawn#:~:text=systemd%2Dnspawn%20is%20like%20the%20chroot%20command%2C%20but%20it%20is%20a%20chroot%20on%20steroids.
Personally want to keep ChromeOS since I don't feel confident in Linux. Might dive into the deep end and use firmware into Chromebook.
Is Systemd-nspawn possible on a Chromebook

**Reply by Qwer-TeX** [reply]
Systemd-nspawn might be possible but you might have to create a chroot inside of the crouton chroot you are running in. In the Crouton chroot bind mount the crouton chroot filesystem with the chroot. I don't know if it will work. I think the chroot would stay mounted if you exit the crouton chroot without shutting of the chromebook.
Or
systemd can be installed in a docker container but isn't recommended. https://stackoverflow.com/questions/51979553/is-it-recommended-to-run-systemd-inside-docker-container
Don't know if that will work.
Or
probably dual boot chromeos alongside linux using [chrx](https://chrx.org/).
That should work and that's what I used.
